### ISSUE TITLE
Weight-uninitialized linear link does not set weight shape correctly when ndim of the first input array is greater than 2.

### ISSUE THREAD

**Issue by delta2323** [chainer/chainer#1614]
The following code does not work.

```
import chainer
from chainer import links as L

import numpy


x = numpy.random.uniform(size=(8, 16, 32)).astype(numpy.float32)
x = chainer.Variable(x)
l = L.Linear(None, 10)
l(x)
```

Traceback

```
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    l(x)
  File "/Users/oonokenta/dev/chainer/chainer/links/connection/linear.py", line 86, in __call__
    return linear.linear(x, self.W, self.b)
  File "/Users/oonokenta/dev/chainer/chainer/functions/connection/linear.py", line 79, in linear
    return LinearFunction()(x, W, b)
  File "/Users/oonokenta/dev/chainer/chainer/function.py", line 123, in __call__
    self._check_data_type_forward(in_data)
  File "/Users/oonokenta/dev/chainer/chainer/function.py", line 199, in _check_data_type_forward
    type_check.InvalidType(e.expect, e.actual, msg=msg), None)
  File "/Users/oonokenta/.pyenv/versions/pyenv-2.7.10/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
chainer.utils.type_check.InvalidType: 
Invalid operation is performed in: LinearFunction (Forward)

Expect: prod(in_types[0].shape[1:]) == in_types[1].shape[1]
Actual: 512 != 16
```

**Comment by okuta** [comment]
This isuue will fix at #1551.

**Comment by delta2323** [comment]
I checked PR #1551 and found that it does not include the code that fixes the issue, if I do not missed something. You mean you are planning to add the implementation to the PR (or to send another PR that fixes the issue)?

**Comment by delta2323** [comment]
I got it. But I think it is better to extract the code that corresponds to this bug because the PR tries to solve two related but different issues. But I think it is OK not to do so if the PR is also almost done. (Personally, I would like to add tests to confirm the bug is surely removed).

**Comment by delta2323** [comment]
Solved via #1551
